sw/device/silicon_creator/rom_ext/data/rom_ext_e2e_testplan.hjson

```diff
@@ -216,5 +216,250 @@
+      desc:
+        '''
+        Verify that ROM_EXT accepts an owner configuration update.
+
+        - Start in an locked & owned state with the `fake` owner as owner.
+        - Unlock the chip into the LockedUpdate state using the `fake` owner unlock key.
+        - Upload an new owner block using the `fake` owner key (ie: embed the `dummy` app key in the `fake` owner block).
+        - Activate ownership with the `fake` activate key.
+        - Confirm code execution with the `dummy` owner application key.
+        '''
+      tags:
+      [
+        rom_ext
+        fpga
+      ]
+      stage: V3
+      tests: []
+    }
+    {
+      name: rom_ext_e2e_bad_locked_update_test
+      desc:
+        '''
+        Verify that ROM_EXT rejects an owner configuration update.
+
+        - Start in an locked & owned state with the `fake` owner as owner.
+        - Unlock the chip into the LockedUpdate state using the `fake` owner unlock key.
+        - Upload an new owner block using the wrong owner key (ie: use the `dummy` owner and app key).
+        - Activate ownership with the `fake` activate key.
+        - Confirm the activate is rejected with an `OwnershipInvalidInfoPage` error.
+        - Confirm that code execution with the new app key is denied.
+        '''
+      tags:
+      [
+        rom_ext
+        fpga
+      ]
+      stage: V3
+      tests: []
+    }
+    {
+      name: rom_ext_e2e_rescue_limit_test
+      desc:
+        '''
+        Verify that ROM_EXT rescue respects the limits set in the owner configuration.
+
+        - Start in an locked & owned state with the `fake` owner as owner.
+        - Unlock the chip into the UnlockedAny state using the `fake` owner unlock key.
+        - Upload an owner block for the `dummy` owner that contains a rescue flash limit.
+        - Activate ownership with the `dummy` activate key.
+        - Upload a test pattern to the flash that would write data past the limit.
+        - Confirm the test pattern exists before the limit, but not after.
+        '''
+      tags:
+      [
+        rom_ext
+        fpga
+      ]
+      stage: V3
+      tests: []
+    }
+    {
+      name: rom_ext_e2e_rescue_permission_test
+      desc:
+        '''
+        Verify that ROM_EXT rescue rejects commands not in the allow list.
+
+        - Start in an locked & owned state with the `fake` owner as owner.
+        - Unlock the chip into the UnlockedAny state using the `fake` owner unlock key.
+        - Upload an owner block for the `dummy` owner that contains a rescue configuration omitting some commands.
+        - Activate ownership with the `dummy` activate key.
+        - Attempt to use rescue to execute the forbidden commands.
+        - Confirm that the rescue protocol rejects the forbidden commands.
+        '''
+      tags:
+      [
+        rom_ext
+        fpga
+      ]
+      stage: V3
+      tests: []
+    }
+    {
+      name: rom_ext_e2e_flash_permission_test
```

Comment:
Should we add corner cases here to cover:

1. An attempt to override settings for a creator partition.
2. Switching from non-scrambled to scrambled + ECC (and any other interesting permutations). This is to show that the implementation will provide a mechanism for the silicon owner to complete the ownership transfer without requiring bootstrap.

Maybe the two cases above are worth to keep in separate test cases.

Reply:
I'll add a test to check that a creator page can't be reconfigured by a bad/malicious owner configuration.

This test (flash_permission_test) supplies a flash configuration that changes the scrambling/ecc page configuration.  I'll amend the text to explicitly state that.